### PR TITLE
ETFE-1238: Updated error validation

### PR DESCRIPTION
**Pull request by r-melvin**

### CODE REVIEW

File: app/controllers/ItemShortageOrExcessController.scala [hunk 1]
```diff
@@ -68,34 +67,46 @@ class ItemShortageOrExcessController @Inject()(
...
-                val formWithError =
-                  formProvider()
+              case value if (value.wrongWithItem == Shortage) && (value.amount > maxAmount(idx, item, isPartiallyRefusingAnAmountOfItem)) =>
+                if (isPartiallyRefusingAnAmountOfItem) {
+                  val formWithError = formProvider()
+                    .fill(value)
+                    .withError(FormError(
+                      key = "amount",
+                      message = "itemShortageOrExcess.amount.error.shortageExceedsRemainder",
+                      args = Seq(value.amount, maxAmount(idx, item, isPartiallyRefusingAnAmountOfItem))
```
Comment (MJCallahanPage) [inline app/controllers/ItemShortageOrExcessController.scala:84]: Where it currently uses: `value.amount` this needs to change. It should be the `RefusedAmount` that is passed into this.

Comment (MJCallahanPage) [reply app/controllers/ItemShortageOrExcessController.scala:84]: I also wonder if the `UnitOfMeasure` needs to be passed to the view. Might be worth checking with clare, as this doesn't really make sense to me:
image

Saying:
> "You have refused 1 litre of this item. The amount of shortage cannot be more than 0"
> "You have refused 2 litres of this item. The amount of shortage cannot be more than 0"
> "You have refused 10 kilograms of this item. The amount of shortage cannot be more than 45"
etc...

Might make more sense? 🤷

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]

File: app/forms/package.scala [hunk 2]
```diff
@@ -16,8 +16,12 @@
...
   private[forms] val MAX_LENGTH_15 = 15
+  private[forms] val MIN_VALUE_0 = 0
   private[forms] val TEXTAREA_MAX_LENGTH = 350
   private[forms] val ALPHANUMERIC_REGEX = "^(?s)(?=.*[A-Za-z0-9]).{1,}$"
   private[forms] val XSS_REGEX = "^(?s)(?!.*javascript)(?!.*[<>;:]).{1,}$"
-  private[forms] val NUMERIC_15_3DP_REGEX: String = "^[1-9]\\d{0,14}$|^([1-9]\\d{0,13}|0)\\.[0-9]$|^([1-9]\\d{0,12}|0)\\.\\d[0-9]$|^([1-9]\\d{0,11}|0)\\.\\d\\d[0-9]$"
+  private[forms] val NUMERIC_15_3DP_REGEX: String =
+    "^[1-9]\\d{0,14}$|^([1-9]\\d{0,13}|0)\\.[0-9]$|^([1-9]\\d{0,12}|0)\\.\\d[0-9]$|^([1-9]\\d{0,11}|0)\\.\\d\\d[0-9]$"
+
+  private[forms] val NUMERIC_REGEX: String = "^(([0-9]*)|(([0-9]*)\\.([0-9]*)))$" // This regex does not accept comma separated numbers e.g. 2,000,000
```
[the remark this answers is not in the document]
Comment (r-melvin) [reply app/forms/package.scala:26]: Probably but I couldn't make it work for a comma every 4th digit nor could I find one that worked properly online.

**Review by MJCallahanPage — APPROVED** [review]
LGTM 🟢 🚀

**Pull request merged**